### PR TITLE
FE: Show a checkbox near parameters in Action Editor Form

### PR DESCRIPTION
**Pull request by uladzimirdev**
Epic: https://github.com/metabase/metabase/issues/30016

### Description

Adds ability to hide action fields from ActionCreator form for both implicit and query actions

### How to verify

- Go to model -> actions -> edit action or `new +` -> create action 
- uncheck "show field", save
- open action again - show field should be unchecked

### Demo

https://www.loom.com/share/8f445ffb8e7e4eb2b631f21dfd991140

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: frontend/src/metabase/actions/containers/ActionCreator/ActionContext/ImplicitActionContextProvider/ImplicitActionContextProvider.tsx [hunk 1]
```diff
@@ -33,12 +33,11 @@ function EditorBody() {
...
 }
 
+// TODO: @uladzimir probably we can delete it, I'm not sure only can we get rid of getDefaultFormSettings
 function cleanFormSettings(formSettings?: ActionFormSettings) {
   const formSettingsWithDefaults = getDefaultFormSettings(formSettings);
 
-  // For implicit actions fields are generated dynamically according to the current schema
-  // For now, we don't let to customize the form to avoid dealing with fields getting out of sync
-  return _.omit(formSettingsWithDefaults, "fields");
+  return formSettingsWithDefaults;
```
Comment (uladzimirdev) [inline frontend/src/metabase/actions/containers/ActionCreator/ActionContext/ImplicitActionContextProvider/ImplicitActionContextProvider.tsx:40]: from now we do not remove `fields` and sync them with BE

File: frontend/src/metabase/actions/containers/ActionCreator/FormCreator/FormFieldEditor/FormFieldEditor.tsx [hunk 2]
```diff
@@ -66,6 +66,7 @@ function FormFieldEditor({
 }: FormFieldEditorProps) {
   const fieldTypeOptions = useMemo(getFieldTypes, []);
   const inputTypeOptions = useMemo(getInputTypes, []);
+  const [hidden, setHidden] = useState(fieldSettings?.hidden ?? false);
```
Comment (ranquild) [inline frontend/src/metabase/actions/containers/ActionCreator/FormCreator/FormFieldEditor/FormFieldEditor.tsx:69]: `const hidden = fieldSettings?.hidden ?? false`

File: frontend/src/metabase/actions/containers/ActionCreator/FormCreator/FormFieldEditor/FormFieldEditor.tsx [hunk 3]
```diff
@@ -97,6 +98,14 @@ function FormFieldEditor({
     });
   };
 
+  const handleVisibilityChange = ({ hidden }: { hidden: boolean }) => {
+    setHidden(hidden);
```
Comment (ranquild) [inline frontend/src/metabase/actions/containers/ActionCreator/FormCreator/FormFieldEditor/FormFieldEditor.tsx:102]: It seems we can (and should) remove the local state

File: frontend/src/metabase/actions/components/ActionFormFieldWidget/ActionFormFieldWidget.tsx [hunk 4]
```diff
@@ -31,15 +32,36 @@ const WIDGETS: Record<InputComponentType, React.FunctionComponent<any>> = {
...
-  ref: React.Ref<any>,
+  { formField, hidden, onVisibilityChange }: FormWidgetProps,
+  ref: Ref<any>,
 ) {
   const Widget =
     (formField.type ? WIDGETS[formField.type] : FormInputWidget) ??
     FormInputWidget;
 
-  return <Widget {...formField} nullable ref={ref} />;
+  const actions =
```
Comment (ranquild) [inline frontend/src/metabase/actions/components/ActionFormFieldWidget/ActionFormFieldWidget.tsx:47]: This looks wrong from the point of composition of components. I understand that `ActionFormFieldWidget` is used in `ActionForm` and `FormFieldEditor` and we only want to get a checkbox in `FormFieldEditor`, that's why we add this `undefined` check. However it seems that we could add `actions` as a prop to `ActionFormFieldWidget` and add a checkbox in `FormFieldEditor`, and don't have the strange `undefined` check

**Review by ranquild — COMMENTED** [review]
Looks good, left a couple of comments

File: frontend/src/metabase/actions/components/ActionFormFieldWidget/ActionFormFieldWidget.tsx (hunk 4 above)
Comment (uladzimirdev) [reply frontend/src/metabase/actions/components/ActionFormFieldWidget/ActionFormFieldWidget.tsx:47]: good point, I think I found the case when we don't need actions only after testing of the solution and it was the simplest fix 😄 I'll change it

File: frontend/src/metabase/actions/containers/ActionCreator/ActionContext/ImplicitActionContextProvider/ImplicitActionContextProvider.tsx (hunk 1 above)
Comment (uladzimirdev) [reply frontend/src/metabase/actions/containers/ActionCreator/ActionContext/ImplicitActionContextProvider/ImplicitActionContextProvider.tsx:40]: @kulyk could you please confirm if we sync settings with BE, we don't need to remove `fields` from settings? /cc @iethree 

(I'm tagging both of you as you worked on actions before)

File: frontend/src/metabase/actions/components/ActionFormFieldWidget/ActionFormFieldWidget.tsx (hunk 4 above)
Comment (uladzimirdev) [reply frontend/src/metabase/actions/components/ActionFormFieldWidget/ActionFormFieldWidget.tsx]: fixed, thanks for a good point!

File: frontend/src/metabase/actions/containers/ActionCreator/FormCreator/FormFieldEditor/FormFieldEditor.tsx (hunk 3 above)
Comment (uladzimirdev) [reply frontend/src/metabase/actions/containers/ActionCreator/FormCreator/FormFieldEditor/FormFieldEditor.tsx]: you're right, it's much easier

File: frontend/src/metabase/actions/containers/ActionCreator/FormCreator/FormFieldEditor/FormFieldEditor.tsx (hunk 2 above)
Comment (uladzimirdev) [reply frontend/src/metabase/actions/containers/ActionCreator/FormCreator/FormFieldEditor/FormFieldEditor.tsx]: fixed

**Review by ranquild — APPROVED** [review]

**Pull request merged**